src/test/java/com/ironcorelabs/tenantsecurity/kms/v1/DevIntegrationTest.java

```diff
@@ -192,6 +192,30 @@ public void roundTripAzureKMSTest() throws Exception {
         assertEqualBytes(decryptedValuesMap.get("doc3"), documentMap.get("doc3"));
     }
 
+    public void roundTripRekeyAWSToAzure() throws Exception {
+        DocumentMetadata metadata = getRoundtripMetadata(this.AWS_TENANT_ID);
+        Map<String, byte[]> documentMap = getRoundtripDataToEncrypt();
+
+        CompletableFuture<PlaintextDocument> roundtrip = getClient().thenCompose(client -> {
+            try {
+                return client.encrypt(documentMap, metadata).thenCompose(encryptedResults -> {
+                    return client.rekeyDocument(encryptedResults, metadata, this.AZURE_TENANT_ID)
+                            .thenCompose(rekeyResults -> {
+                                DocumentMetadata newMetadata = getRoundtripMetadata(this.AZURE_TENANT_ID);
```

Comment:
You could add an `assertEquals` here (I think) to show that the encryptedFields Map hasn't changed... assuming the Map we're using has (deep) `equals` defined, I guess.

Reply:
I like it. Done here and and in the local test